controllers/vspherecluster_reconciler.go

```diff
@@ -499,64 +493,72 @@ func (r *clusterReconciler) controlPlaneMachineToCluster(ctx context.Context, o
 
 	vsphereMachine, ok := o.(*infrav1.VSphereMachine)
 	if !ok {
-		log.Error(nil, fmt.Sprintf("expected a VSphereMachine but got a %T", o))
+		log.Error(nil, fmt.Sprintf("Expected a VSphereMachine but got a %T", o))
 		return nil
 	}
 	log = log.WithValues("VSphereMachine", klog.KObj(vsphereMachine))
 	ctx = ctrl.LoggerInto(ctx, log)
 
 	if !infrautilv1.IsControlPlaneMachine(vsphereMachine) {
+		log.V(6).Info("Skipping VSphereCluster reconcile as Machine is not a control plane Machine")
 		return nil
 	}
+
 	if len(vsphereMachine.Status.Addresses) == 0 {
+		log.V(6).Info("Skipping VSphereCluster reconcile as Machine does not have an IP address")
 		return nil
 	}
+
 	// Get the VSphereMachine's preferred IP address.
 	if _, err := infrautilv1.GetMachinePreferredIPAddress(vsphereMachine); err != nil {
-		if err == infrautilv1.ErrNoMachineIPAddr {
+		if errors.Is(err, infrautilv1.ErrNoMachineIPAddr) {
+			log.V(6).Info("Skipping VSphereCluster reconcile as Machine does not have a preferred IP address")
 			return nil
 		}
-		log.Error(err, "Failed to get preferred IP address for VSphereMachine")
+		log.V(4).Error(err, "Failed to get preferred IP address for VSphereMachine")
 		return nil
 	}
 
 	// Fetch the CAPI Cluster.
 	cluster, err := clusterutilv1.GetClusterFromMetadata(ctx, r.Client, vsphereMachine.ObjectMeta)
 	if err != nil {
-		log.Error(err, "VSphereMachine is missing cluster label or cluster does not exist")
+		log.V(4).Error(err, "VSphereMachine is missing cluster label or cluster does not exist")
 		return nil
 	}
 	log = log.WithValues("Cluster", klog.KObj(cluster))
+	if cluster.Spec.InfrastructureRef != nil {
+		log = log.WithValues("VSphereCluster", klog.KRef(cluster.Namespace, cluster.Spec.InfrastructureRef.Name))
+	}
 	ctx = ctrl.LoggerInto(ctx, log)
 
 	if conditions.IsTrue(cluster, clusterv1.ControlPlaneInitializedCondition) {
+		log.V(6).Info("Skipping VSphereCluster reconcile as control plane is already initialized")
 		return nil
 	}
 
 	if !cluster.Spec.ControlPlaneEndpoint.IsZero() {
+		log.V(6).Info("Skipping VSphereCluster reconcile as Cluster control plane endpoint is already set")
 		return nil
 	}
 
 	if cluster.Spec.InfrastructureRef == nil {
-		log.Info("Failed to get VSphereCluster: Cluster.spec.infrastructureRef is not yet set")
+		log.Error(nil, "Failed to get VSphereCluster: Cluster.spec.infrastructureRef is not yet set")
```

Comment:
As far as I can tell it's fine as is:

> // The msg argument should be used to add context to any underlying error,
> // while the err argument should be used to attach the actual error that
> // triggered this log line, if present. The err parameter is optional
> // and nil may be passed instead of an error instance.